Feature/tournament preparation

Kneshal opened the pull request:
Новый функционал - кнопки для удаления любых пользователей и добавления прокси ботов (если количество зарегистрированных участников нечетное). Организатор турнира может вручную выбрать аккаунт прокси бота из списка, либо просто нажать на кнопку добавить и в списке появится первый прокси-бот, который не участвует в этом турнире.
!!! Для работы обязательно нужно заранее создать хотя-бы один аккаунт прокси бота из панели администратора - в нем следует сделать активным поле proxy-bot.
Также поправил отображение таблицы участников.

Kneshal committed:
tournament preparation
- add field 'proxy_bot' to Profile model;
- add property 'reg_is_open_or_close' to Tournament model;
- edit tournament_detail template - new buttons;
- add new url to urls.py;
- change context in tournament_detail_view (new quantity flag);
- add new file with players manage views;
- add 2 functions for add/delete proxy_bot from tournament.

Kneshal committed:
Add special icon to delete players from tt

Kneshal committed:
manual method to add a bot:
- add new form with choise field;
- edit view player_add_bot (user can choose manual or auto methods);
- add form to tournament_detail_view;
- edit template tournament_detail.html.

lejbron commented on tournaments/models.py:
```diff
@@ -144,6 +144,18 @@ def save(self, *args, **kwargs):
             self.tt_slug = slugify(self)
         super(Tournament, self).save(*args, **kwargs)
 
+    @property
+    def reg_is_open_or_close(self):
+        if self.tt_status == 'reg' or self.tt_status == 'creg':
+            return True
+        return False
+
+    @property
+    def is_act_or_fin(self):
+        if self.tt_status == 'act' or self.tt_status == 'fin':
+            return True
+        return False
+
```
По идее if/else не нужны, достаточно просто возвращать условные выражения, они в итоге и вернут True или False.
Так же думаю, что лучше сделать по такому свойству для каждого статуса отдельно. Чаще всего пока действительно нужно учитывать один или второй, но может возникнуть ситуация, когда нужно будет проверить только один статус, и тогда придется делать дополнительное свойство.

lejbron commented on tournaments/templates/tournament_detail.html:
```diff
@@ -7,21 +7,58 @@
 {% endif %}
+
+<div class="row">
+  <div class="col-sm-4">
+    <table class="table table-striped table-sm">
```
Нужно добавить условие на отображение - должно появляться только после открытия регистрации и после того, как зарегистрировался хотя бы один игрок - чтобы не отображать пустую таблицу.

lejbron commented on tournaments/templates/tournament_detail.html:
```diff
@@ -7,21 +7,58 @@
+            <a href="{% url 'player-detail' record.player.id %}">
+              {{ record }}
+            </a>
+            {% if tournament.superviser == user %}
+            <a href="{% url 'player_delete' tournament.tt_slug record.player.id %}" role="button">
+              <i class="fas fa-user-times ml-1"></i>
```
Нужно запрашивать доп подтверждение на удаление - например, модальное окно "Вы действительно хотите убрать участника из турнира?"
Формулировку можно изменить.

lejbron reviewed (changes requested):
Необходимо исправить действие кнопки Закрыть регистрацию в случае нечетного количества игроков. 
Сейчас после нажатия на эту кнопку открывается url:
 .../tournaments/<tt_slug>/close_registration с модальным окном предупреждения, НО на этой странице нет формы добавления прокси. 

Этот момент как раз хотел с тобой обсудить голосом, но так и не созвонились в тот день.
По факту проблема в том, что функция render открывает другой url, а функция redirect не обновляет context. Нужно подумать, как исправить.

Kneshal committed:
edit tournament_detail template (new table)

Kneshal committed:
hotfix for tournament_detail template:
- simplified conditions for displaying content;
- add new property to Tournament model.

Kneshal committed:
hotfix after code review:
- add new modal template for delete player;
- edit properties in Tournament model;
- edit tournament_detail template;
- fix bug in add_proxy_bot view.

Kneshal committed:
hotfix for close registration button:
- change flag name in status_change_forms to 'odd_number_flag';
- edit flag in tournament_card.html;
- add proxy_bot_form to tournament_detail_view.
